linc_convert/modalities/lsm/transfer.py

```diff
@@ -0,0 +1,106 @@
+"""
+Convert a collection of spool .dat files generated by light sheet microscopy to compressed tar files and upload with the DANDI client.
+"""
+
+# stdlib
+import os
+from datetime import datetime
+from pathlib import Path
+
+# externals
+import tarfile
+import cyclopts
+import dandi.download
+import dandi.upload
+
+# internals
+from linc_convert.modalities.lsm.cli import lsm
+
+transfer = cyclopts.App(name="transfer", help_format="markdown")
+lsm.command(transfer)
+
+@transfer.default
+def dandi_transfer(input_dir, dandiset_url, dandi_instance, subject, output_dir='.',  max_size_gb=2.00, upload=False):
+    """
+    Upload .dat files to DANDI in batched, compressed tar archives.
+    
+    Parameters
+    ----------
+    input_dir : str
+        Directory containing .dat files to upload
+    dandiset_url : str
+        URL for the dandiset to upload (e.g., https://lincbrain.org/dandiset/000010)
+    dandi_instance : str
+        DANDI server (e.g. linc, dandi)
+    output_dir : str, optional
+        Directory to save the Dandiset directory (default: '.')
```

Comment:
We delete everyting in the end, right? Should we just use a `tempfile.TemporaryDirectory` instead of asking the user for a location?